src/opensignals/data/provider.py

```diff
@@ -0,0 +1,257 @@
+from abc import ABC, abstractmethod
+from concurrent import futures
+from datetime import datetime, date, time
+import logging
+import shutil
+
+from dateutil.relativedelta import relativedelta, FR
+import pandas as pd
+from tqdm import tqdm
+
+logger = logging.getLogger(__name__)
+
+AWS_BASE_URL = 'https://numerai-signals-public-data.s3-us-west-2.amazonaws.com'
+SIGNALS_UNIVERSE = f'{AWS_BASE_URL}/latest_universe.csv'
+SIGNALS_TICKER_MAP = f'{AWS_BASE_URL}/signals_ticker_map_w_bbg.csv'
+SIGNALS_TARGETS = f'{AWS_BASE_URL}/signals_train_val_bbg.csv'
+
+
+class Provider(ABC):
+    """Common base class for (daily) stock price data"""
+
+    @staticmethod
+    def get_tickers():
+        ticker_map = pd.read_csv(SIGNALS_TICKER_MAP)
+        ticker_map = ticker_map.dropna(subset=['yahoo'])
+        logger.info(f'Number of eligible tickers: {ticker_map.shape[0]}')
+
+        if ticker_map['yahoo'].duplicated().any():
+            num = ticker_map["yahoo"].duplicated().values().sum()
+            raise Exception(
+                f'Found duplicated {num}'
+                ' yahoo tickers'
+            )
+
+        if ticker_map['bloomberg_ticker'].duplicated().any():
+            num = ticker_map["bloomberg_ticker"].duplicated().values().sum()
+            raise Exception(
+                f'Found duplicated {num}'
+                ' bloomberg_ticker tickers'
+            )
+
+        return ticker_map
+
+    @staticmethod
+    def get_ticker_data(db_dir):
+        ticker_data = pd.DataFrame({
+            'bloomberg_ticker': pd.Series([], dtype='str'),
+            'date': pd.Series([], dtype='datetime64[ns]')
+        })
+        if len(list(db_dir.rglob('*.parquet'))) > 0:
```

Comment:
should we prefix the parquet files somehow? To ensure these are opensignals files.

Reply:
I don't think we should asume there is a reason to mix the raw data files for opensignals with some other data used. The way I use this library is that I hav a folder the raw price data downloaded by opensignals (e.g price/yahoo) and for additional data  I have a similar structure <data type>/<data source>.